qiskit/pulse/instructions/directives.py

```diff
@@ -55,3 +55,80 @@ def channels(self) -> Tuple[chans.Channel]:
     def __eq__(self, other):
         """Verify two barriers are equivalent."""
         return isinstance(other, type(self)) and set(self.channels) == set(other.channels)
+
+
+class AreaBarrier(Directive):
```

Comment:
Yeah that's being me with poor vocabulary lol `RelativeBarrier` can take list of channels, so this spreads through vertical direction. This this could be something like `HorizontalBarrier` or maybe `Placeholder` instruction, but these are bit vague. Do you have any wording suggestion?

Reply:
How about `TimeBlock`? This indicates that the directive `Blocks` a channel for a `Time`.

Reply:
Nice. Renamed to `TimeBlockade` in https://github.com/Qiskit/qiskit-terra/pull/8949/commits/1f1db8de02baed5abced3c850dc262efce9492f6. I avoided to use `Block` because this usually refers to a chunk of program in this context.